tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/types/package.scala

```diff
@@ -225,14 +228,15 @@ package object types {
   /**
    * A cell constant, that is, just a name that expresses string constants in TLA+.
    */
-  sealed case class ConstT() extends CellT with Serializable {
-    override val signature: String = "str"
+  sealed case class ConstT(utype: String = "") extends CellT with Serializable {
```

Comment:
I would prefer using `utype = "Str"` for the generic strings. This would give you a direct way of using `ModelValueCache`. And you would not need all these if-then-else tests below.

Reply:
I agree, though I'd go with lowercase `str`